NitroxModel/Core/NitroxServiceLocator.cs

```diff
@@ -126,10 +126,10 @@ private static void CheckServiceResolutionViability()
     private static void OnLifetimeScopeEnded() => LifetimeScopeEnded?.Invoke(null, EventArgs.Empty);
 
     /// <summary>
-    ///     Generic static class to cache type with very fast lookups. Only use for singleton types.
+    ///     Generic cache type with very fast lookups. Only use for singleton types.
     /// </summary>
     /// <typeparam name="T">Type in the cache, should be singleton.</typeparam>
-    public static class Cache<T> where T : class
```

Comment:
In what aspects is a ref struct better than a static class? I can't find sufficient explanation online :/

Reply:
I used `struct` here because the JIT has more options for optimization (classes being more complicated objects to the compiler). I ran a benchmark on .NET 7 and structs win slightly for this use case:

```
|              Method |     Mean |    Error |   StdDev |
|-------------------- |---------:|---------:|---------:|
|  BenchmarkWithClass | 19.98 us | 0.059 us | 0.055 us |
| BenchmarkWithStruct | 18.02 us | 0.027 us | 0.024 us |
```
[Benchmark code](https://gist.github.com/Measurity/32748b031fa523728ca5069d8bdd4f77)